fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -900,6 +904,137 @@ void sampledDelete(boolean isSynthetic, boolean isGrouped, long seed) throws IOE
+        // Test that empty partitions are removed during repartitioning
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilder, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPrimaryKeySegmentIndexEnabled(true)
+                .setPartitionHighWatermark(10)
+                .build();
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 2)
+                .addProp(LuceneRecordContextProperties.LUCENE_MERGE_SEGMENTS_PER_TIER, (double)dataModel.nextInt(10) + 2) // it must be at least 2.0
+                .build();
+
+        // Create multiple partitions with documents
+        try (FDBRecordContext context = openContext(contextProps)) {
+            dataModel.saveRecordsToAllGroups(25, context);
+            commit(context);
+        }
+
+        explicitMergeIndex(dataModel.index, contextProps, dataModel.schemaSetup);
+
+        // Verify we have 3 partitions (25 docs) initially
+        dataModel.getPartitionCounts(() -> openContext(contextProps)).forEach((groupingKey, partitionCounts) ->
+                assertThat(partitionCounts, hasSize(3)));
+
+        // Delete documents
+        try (FDBRecordContext context = openContext(contextProps)) {
+            final FDBRecordStore recordStore = dataModel.createOrOpenRecordStore(context);
+            dataModel.groupingKeys().forEach(groupingKey -> {
+                List<Tuple> sortedPartitionKeys = dataModel.groupingKeyToPrimaryKeyToPartitionKey.get(groupingKey)
+                        .values().stream().sorted().collect(Collectors.toList());
+                // delete enough docs to empty partition
+                Set<Tuple> partitionKeysToDelete = new HashSet<>(sortedPartitionKeys.subList(start, end));
+                dataModel.recordsUnderTest().stream()
+                        .filter(rec -> partitionKeysToDelete.contains(rec.getPartitioningKey()))
+                        .forEach(rec -> rec.deleteRecord(recordStore).join());
+            });
+            context.commit();
+        }
+
+        // Before repartitioning, we should still have 3 partitions (including the empty one)
+        dataModel.getPartitionCounts(() -> openContext(contextProps)).forEach((groupingKey, partitionCounts) ->
+                assertThat(partitionCounts, hasSize(3)));
+
+        // Trigger repartitioning - this should remove the empty partition
+        explicitMergeIndex(dataModel.index, contextProps, dataModel.schemaSetup);
+
+        // After repartitioning, we should have only 2 partitions (empty one removed)
+        dataModel.getPartitionCounts(() -> openContext(contextProps)).forEach((groupingKey, partitionCounts) ->
+                assertThat(partitionCounts, hasSize(2)));
```

Comment:
You have going from 3 to 2 partitions and 3 to one empty partition, would it be beneficial to go from 3 to 1 non-empty partition?